scripts/deploy.sh

```diff
@@ -7,17 +7,88 @@ pull_latest() {
 
 rebuild_regs() {
   # Build WCA regulations
-  # Use only one core because we get better backtraces without multiprocessing,
-  # and because we might be on the production server and should not eat up all
-  # cores just building the regulations.
-  regulations/make.py --num-workers 1 --wca
-  regs_folder=WcaOnRails/app/views/regulations
+  # Uses wrc, see here: https://github.com/cubing/wca-regulations-compiler
+  # pdf generations relies on wkhtmltopdf (with patched qt), which should be in $PATH
+  build_folder=regulations/build
+  regs_folder_root=WcaOnRails/app/views
+  regs_folder=$regs_folder_root/regulations
+  regs_version=$regs_folder/version
+  translations_version=$regs_folder/translations/version
+
+  if [ -a $build_folder ]; then
+    rm -rf $build_folder
+  fi
+  mkdir -p $build_folder
+
+  # We want latest commit hash, so we do a shallow copy of the repositories (and not simply a wget)
+  git clone --depth=1 https://github.com/cubing/wca-regulations.git $build_folder/wca-regulations
+  git clone --depth=1 https://github.com/cubing/wca-regulations-translations.git $build_folder/wca-regulations-translations
+  git_reg_hash=`git -C $build_folder/wca-regulations rev-parse --short HEAD`
+  git_translations_hash=`git -C $build_folder/wca-regulations-translations rev-parse --short HEAD`
+
+  rebuild_regulations=1
+  rebuild_translations=1
+  # Check if the cloned regulations match the current version
+  if [ -r $regs_version ] && [ "`cat $regs_version`" == "$git_reg_hash" ]; then
+    rebuild_regulations=0
+  fi
+  # Check if the cloned translations match the current version
+  if [ -r $translations_version ] && [ "`cat $translations_version`" == "$git_translations_hash" ]; then
+    rebuild_translations=0
+  fi
+  if [ $rebuild_regulations -eq 0 ] && [ $rebuild_translations -eq 0 ]; then
+    echo "WCA Regulations and translations are up to date."
+    return
+  fi
+
+  # Else we have to rebuild something
+
+  # This save tracked files that may have unstashed changes too
+  cp -r $regs_folder $build_folder
+
+  # Checkout data (scramble programs, history)
+  git checkout origin/regulations-data $build_folder
+  git reset HEAD $build_folder
```

Comment:
Sure, the `git checkout` part will update the "regulations/build" path to the version in `regulations-data`, without changing branch, updating the index.
Therefore you end up with a lot of file tagged for addition, since they are not present in the current branch.

The default mode for `git reset` is `--mixed`, which does not modify the content of any file, but which will reset the index (to remove the file tagged for addition).
So in this case it means "remove the file marked for addition from the index, but do not do anything to the working tree".
Therefore unstashed changes are preserved :)